packages/blog/src/utils/image.ts

```diff
@@ -16,5 +16,30 @@ export function getImageUrlFromMedia({
   format = 'large',
   fallback = '/images/WElogo.png',
 }: GetImageUrlFromMediaParams) {
-  return media ? media.formats[format].url : fallback || '/images/WElogo.png'
+  try {
+    if (media) {
+      const preferredFormat = media.formats[format]
+      if (preferredFormat) {
+        return preferredFormat.url
+      } else {
+        //search for the following format available
+        const nextAvailableFormatKeys = Object.keys(media.formats) as FormatType[]
+        if (nextAvailableFormatKeys.length > 0) {
+          nextAvailableFormatKeys.forEach(formatKey => {
+            const nextAvailableFormatData = media.formats[formatKey]
+            if (nextAvailableFormatData) {
+              return nextAvailableFormatData.url
+            }
+          })
+          return media.url
+        } else {
+          return media.url
+        }
+      }
+    }
+    return fallback || '/images/WElogo.png'
```

Comment:
This is completely subjective, but just in case it could help... I usually prefer early returns than nested blocks to avoid "[hadoukens](https://external-preview.redd.it/6TTm4tS5pMCiq2KlgviubpGKt1QzvetOi2Z6Z3lB1Nc.jpg?auto=webp&s=c1df961d84c8d76048a93c1d261e425d3737d779)"

(https://medium.com/swlh/return-early-pattern-3d18a41bba8)

```suggestion
  if (!media) return fallback || '/images/WElogo.png'

  const preferredFormat = media.formats[format]
  if (preferredFormat) return preferredFormat.url

  //search for the following format available
  const nextAvailableFormatKeys = Object.keys(media.formats) as FormatType[]
  const nextAvailableFormat = nextAvailableFormatKeys.find(formatKey => media.formats[formatKey])
  if (!nextAvailableFormat) return media.url;

  return nextAvailableFormat.url;
```